daemon/tests/happy_path.rs

```diff
@@ -186,3 +222,68 @@ async fn maker_notices_lack_of_taker() {
         next(maker.connected_takers_feed()).await.unwrap()
     );
 }
+
+/// Hide the implementation detail of arriving at the Cfd open state.
+/// Useful when reading tests that should start at this point.
+/// For convenience, returns also OrderId of the opened Cfd.
+async fn start_from_open_cfd_state() -> (Maker, Taker, OrderId) {
+    let heartbeat_interval = Duration::from_secs(60);
+    let maker_listener = tokio::net::TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let mut maker = Maker::start(
+        &MakerConfig::default().with_heartbeat_interval(heartbeat_interval),
+        maker_listener,
+    )
+    .await;
+    let mut taker = Taker::start(
+        &TakerConfig::default().with_heartbeat_timeout(heartbeat_interval * 2),
+        maker.listen_addr,
+        maker.identity_pk,
+    )
+    .await;
+
+    is_next_none(taker.order_feed()).await.unwrap();
+
+    maker.publish_order(dummy_new_order()).await;
+
+    let (_, received) = next_order(maker.order_feed(), taker.order_feed())
+        .await
+        .unwrap();
+
+    taker.mocks.mock_oracle_announcement().await;
+    maker.mocks.mock_oracle_announcement().await;
+
+    taker.take_order(received.clone(), Usd::new(dec!(5))).await;
+    let (_, _) = next_cfd(taker.cfd_feed(), maker.cfd_feed()).await.unwrap();
+
+    maker.mocks.mock_party_params().await;
+    taker.mocks.mock_party_params().await;
+
+    maker.mocks.mock_monitor_oracle_attestation().await;
+    taker.mocks.mock_monitor_oracle_attestation().await;
+
+    maker.mocks.mock_oracle_monitor_attestation().await;
+    taker.mocks.mock_oracle_monitor_attestation().await;
+
+    maker.mocks.mock_monitor_start_monitoring().await;
+    taker.mocks.mock_monitor_start_monitoring().await;
+
+    maker.accept_take_request(received.clone()).await;
+
+    let (taker_cfd, maker_cfd) = next_cfd(taker.cfd_feed(), maker.cfd_feed()).await.unwrap();
+    assert!(matches!(taker_cfd.state, CfdState::ContractSetup { .. }));
+    assert!(matches!(maker_cfd.state, CfdState::ContractSetup { .. }));
+
+    maker.mocks.mock_wallet_sign_and_broadcast().await;
+    taker.mocks.mock_wallet_sign_and_broadcast().await;
+
+    let (taker_cfd, maker_cfd) = next_cfd(taker.cfd_feed(), maker.cfd_feed()).await.unwrap();
+    assert!(matches!(taker_cfd.state, CfdState::PendingOpen { .. }));
+    assert!(matches!(maker_cfd.state, CfdState::PendingOpen { .. }));
+
+    deliver_lock_finality_event(&maker, &taker, received.id).await;
+
+    let (taker_cfd, maker_cfd) = next_cfd(taker.cfd_feed(), maker.cfd_feed()).await.unwrap();
+    assert!(matches!(taker_cfd.state, CfdState::Open { .. }));
+    assert!(matches!(maker_cfd.state, CfdState::Open { .. }));
```

Comment:
Is this necessary in the setup function?

Reply:
I'd keep this one, as it gives the confidence for users of the function - removed other assertions though!